### PR TITLE
Investment collection refactor

### PR DESCRIPTION
**Pull request by tyom**
This refactor splits core functionality of investment collections into modules which allows better reusability in other parts of the application.

These are under the hood changes. No changes to the UI.

### CODE REVIEW

File: src/apps/investment-projects/builders.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+function buildInvestmentFilters (originalQuery = {}) {
+  const formOptions = {
+    stage: metadataRepo.investmentStageOptions.map(transformObjectToOption),
+    investment_type: metadataRepo.investmentTypeOptions.map(transformObjectToOption),
+    sector: metadataRepo.sectorOptions.map(transformObjectToOption),
+  }
+
+  return Object.keys(collectionFilterLabels.edit).reduce((filtersObj, filterName) => {
+    const options = get(formOptions, filterName, [])
```
Comment (teneightfive) [inline src/apps/investment-projects/builders.js]: Why not just `const options = formOptions.filterName || []`?

File: src/apps/investment-projects/builders.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    { value: 'project_code', label: 'Project code' },
+    { value: 'name:asc', label: 'Project name' },
+    { value: 'stage.name', label: 'Stage' },
+    { value: 'total_investment:desc', label: 'Investment value: high to low' },
+    { value: 'total_investment:asc', label: 'Investment value: low to high' },
+  ]
+
+  const query = Object.assign(
+    {},
+    { sortby: options[0].value }, // default
```
Comment (teneightfive) [inline src/apps/investment-projects/builders.js]: Do you need the comment?

File: src/apps/investment-projects/views/list.njk [hunk 3]
```diff
@@ -5,52 +5,52 @@
 {% block main_grid_left_column %}
   {{
     ResultsFilters({
-      selectedSorting: form.data.sorting,
+      sort: sort,
```
Comment (teneightfive) [inline src/apps/investment-projects/views/list.njk]: `sort,`?

Comment (teneightfive) [inline src/apps/investment-projects/views/list.njk]: Ignore me. It's a `.njk` file.

File: src/apps/investment-projects/builders.js (hunk 2 above)
Comment (tyom) [inline src/apps/investment-projects/builders.js]: Probably not, just wanted to make sure it's clear.

**Pull request merged**